yellowbrick/text/dispersion.py

```diff
@@ -95,34 +120,91 @@ def fit(self, text):
         text : list
             Should be provided as a list of documents that contain
             a list of words in the order they appear in the document.
+
+        y : ndarray or Series of length n
+            An optional array or series of target or class values for
+            instances. If this is specified, then the points will be colored
+            according to their class.
+
+        kwargs : dict
+            Pass generic arguments to the drawing method
+
+        Returns
+        -------
+        self : instance
+            Returns the instance of the transformer/visualizer
         """
 
+        if y is not None:
+            self.classes_ = np.unique(y)
+        elif y is None and self.labels is not None:
+            self.classes_ = np.array([self.labels[0]])
+        else:
+            self.classes_ = np.array([self.NULL_CLASS])
+
         # Create an index (e.g. the y position) for the target words
         self.target_words_ = np.flip(self.words, axis=0)
         if self.ignore_case:
             self.target_words_ = np.array([w.lower() for w in self.target_words_])
 
         # Stack is used to create a 2D array from the generator
-        points = np.stack(self._compute_dispersion(text))
-        self.draw(points)
+        self.points_target = np.stack(self._compute_dispersion(text, y))
+
+        self.points = np.stack(zip(self.points_target[:,0].astype(int),
+                               self.points_target[:,1].astype(int)))
+
+        self.target = self.points_target[:,2]
+
+        self.draw(self.points, self.target)
         return self
 
-    def draw(self, points, **kwargs):
+    def draw(self, points, target=None, **kwargs):
         """
         Called from the fit method, this method creates the canvas and
-        draws the distribution plot on it.
+        draws the plot on it.
         Parameters
         ----------
         kwargs: generic keyword arguments.
         """
 
+        # Resolve the labels with the classes
+        labels = self.labels if self.labels is not None else self.classes_
+        if len(labels) != len(self.classes_):
```

Comment:
Might be a good idea to add a test that make sure this error gets raised in the number of labels doesn't match the number of colors.

Reply:
Maybe I am missing something very obvious.  Won't the number of labels always match the number of color since the number of colors is dictated by the number of labels.  L179-181

Reply:
Sorry - I meant to type "if the number of labels doesn't match the number of classes". I am referring to you `if` statement on line 172. I was suggesting that you add a test that ensures the error is raised if the number of labels doesn't match the number of classes.

Reply:
I believe I am already catching this mismatch in my tests...  [L115-L129](https://github.com/DistrictDataLabs/yellowbrick/blob/5d6da64aefaa47208fe36db5691ae1200540d3d0/tests/test_text/test_dispersion.py#L115-L129)

Reply:
How about using a `match` inside your test to ensure that the custom error message you have written for the `YellowbrickValueError` is raised. Something like [this](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/tests/test_classifier/test_classification_report.py#L166-L172) or [this](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/tests/test_draw.py#L33-L34).

Reply:
Got it!